src/Cache_Command.php

```diff
@@ -20,34 +18,130 @@ class Cache_Command extends WP_CLI_Command {
 	 *
 	 * ## OPTIONS
 	 *
-	 * <post_id>
+	 * [<post_id>]
 	 * : ID of the post to clear the cache for.
 	 *
+	 * [--all]
+	 * : Clear all oEmbed caches.
+	 *
 	 * ## EXAMPLES
 	 *
 	 *     # Clear cache for a post
 	 *     $ wp embed cache clear 123
 	 *     Success: Cleared oEmbed cache.
+	 *
+	 *     # Clear all caches
+	 *     $ wp embed cache clear --all
+	 *     Success: Cleared all oEmbed caches.
 	 */
 	public function clear( $args, $assoc_args ) {
 		/** @var \WP_Embed $wp_embed */
-		global $wp_embed;
+		global $wp_embed, $wpdb;
+
+		$all = \WP_CLI\Utils\get_flag_value( $assoc_args, 'all' );
+
+		$post_id = isset( $args[0] ) ? $args[0] : null;
+
+		if ( null === $all && null === $post_id ) {
+			WP_CLI::error( 'You must specify at least one post id or use --all' );
+		}
+
+		if ( $post_id && $all ) {
+			WP_CLI::error( 'You cannot specify both a post id and use --all' );
+		}
+
+		// Delete all oEmbed caches.
+		if ( $all ) {
+			// Get post meta oEmbed caches
+			$oembed_post_meta_post_ids = (array) $wpdb->get_col(
+				"SELECT DISTINCT post_id FROM $wpdb->postmeta
+				WHERE meta_key REGEXP '^_oembed_[0-9a-f]{32}$'
+				OR meta_key REGEXP '^_oembed_time_[0-9a-f]{32}$'"
+			);
+			// Get posts oEmbed caches
+			$oembed_post_post_ids = (array) $wpdb->get_col(
+				"SELECT ID FROM $wpdb->posts
+				WHERE post_type = 'oembed_cache'
+				AND post_status = 'publish'
+				AND post_name REGEXP '^[0-9a-f]{32}$'"
+			);
+
+			// Get transient oEmbed caches
+			$oembed_transients = $wpdb->get_col(
+				"SELECT option_name FROM $wpdb->options
+				WHERE option_name REGEXP '^_transient_oembed_[0-9a-f]{32}$'"
+			);
+
+			$oembed_caches = array(
+				'post' => $oembed_post_meta_post_ids,
+				'oembed post' => $oembed_post_post_ids,
+				'transient' => $oembed_transients,
+			);
 
-		$post_id = $args[0];
+			$total = array_sum( array_map( function( $items ) {
+				return count( $items );
+			}, $oembed_caches ) );
+
+			// Delete post meta oEmbed caches
+			foreach ( $oembed_post_meta_post_ids as $post_id ) {
+				$wp_embed->delete_oembed_caches( $post_id );
+			}
+
+			// Delete posts oEmbed caches
+			foreach ( $oembed_post_post_ids as $post_id ) {
+				wp_delete_post( $post_id, true );
+			}
+
+			// Delete transient oEmbed caches
+			foreach ( $oembed_transients as $option_name ) {
+				delete_transient( str_replace( '_transient_', '', $option_name ) );
+			}
+
+			if ( $total > 0 ) {
+				$details = array();
+				foreach ( $oembed_caches as $type => $items ) {
+					$count = count( $items );
+					$details[] = sprintf(
+						'%1$d %2$s %3$s',
+						$count,
+						$type,
+						Utils\pluralize( 'cache', $count )
+					);
+				}
+
+				$message = sprintf(
+					'Cleared %1$d oEmbed %2$s: %3$s.',
+					$total,
+					Utils\pluralize( 'cache', $total ),
+					implode( ', ', $details )
+				);
+
+				WP_CLI::success( $message );
+			} else {
+				WP_CLI::error( 'No oEmbed caches to clear!' );
```

Comment:
```suggestion
				WP_CLI::error( 'No oEmbed caches to clear.' );
```

Should this be `WP_CLI::success()`?

Reply:
[Currently](https://github.com/wp-cli/embed-command/blob/3987e2051354eaad842c8612ea9255493534c589/src/Cache_Command.php#L50) is `WP_CLI::error();`, I stuck to what's in place right now.

But agree on the change. Makes more sense. You tell me.